Wikipedia/Code/NotificationsCenterDetailViewModel+ActionExtensions.swift

```diff
@@ -124,6 +124,18 @@ extension NotificationsCenterDetailViewModel {
         }
         return secondaryActions
     }
+
+    /// Do not include secondary action if its destination is the same as the primary action
+    var uniqueSecondaryActions: [NotificationsCenterAction] {
+        guard let primaryActionURL = primaryAction?.actionData?.url else {
+            return secondaryActions
+        }
+
+        return secondaryActions.filter({ action in
+            action.actionData?.url != primaryActionURL
+        })
+    }
```

Comment:
Interesting! I didn't consider duplicates within the secondary actions themselves.

Keeping this at the post-processing level, I just added `Hashable` conformance to `NotificationsCenterIconType` to be able to use Set to remove duplicates from both the secondary actions and the sheet actions. Does this solution feel sufficient to you?

I think it's ready for another look!

Reply:
@staykids thanks! This approach sounds fine to me - though the deduping didn't work for me until I also conformed `NotificationsCenterAction` and `NotificationsCenterActionData` to `Hashable`. So just that one last change is needed I think.

Reply:
Strange! I must be misunderstanding or missing something, because I expected that all the nested `struct`'s properties conforming to `Equatable` would be sufficient.

Updated! Also updated the tests to reflect the expected sheet actions after removing duplicates.

Reply:
No problem. I'm actually a little surprised it worked with `NSOrderedSet` at all, I thought you'd have to use a more Objective-C friendly object 😅. Seems to be working well though!